### PR TITLE
remove todo(#42787)Add fallback for cronjobs, and move some useful funct…

### PR DESCRIPTION
**Pull request by zjj2wry**
…ion to cmdutil

**What this PR does / why we need it**:
remove todo
relation with #42787

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by xiangpengzhao** [comment]
@zjj2wry please fix this failure.

```
W0818 08:31:29.917] Use --sandbox_debug to see verbose messages from the sandbox.
W0818 08:31:29.919] 2017/08/18 08:31:29 missing strict dependencies:
W0818 08:31:29.925] 	pkg/kubectl/cmd/util/factory_client_access.go: import of k8s.io/api/apps/v1beta1, which is not a direct dependency
W0818 08:31:29.928] 	pkg/kubectl/cmd/util/factory_client_access.go: import of k8s.io/api/batch/v2alpha1, which is not a direct dependency
W0818 08:31:29.929] Use --strategy=GoCompile=standalone to disable sandboxing for the failing actions.
W0818 08:31:30.051] ____Building complete.
W0818 08:31:30.116] ____Elapsed time: 128.831s, Critical Path: 55.78s
W0818 08:31:30.124] Run: ('/workspace/./test-infra/jenkins/../scenarios/../images/pull_kubernetes_bazel/coalesce.py',)
I0818 08:31:30.174] Build failed
E0818 08:31:30.175] Command failed
I0818 08:31:30.175] process 474 exited with code 1 after 3.0m
E0818 08:31:30.176] FAIL: pull-kubernetes-bazel
```

**Comment by zjj2wry** [comment]
@xiangpengzhao thanks

File: pkg/kubectl/cmd/util/factory_client_access.go [hunk 1]
```diff
@@ -553,6 +556,53 @@ func DefaultGenerators(cmdName string) map[string]kubectl.Generator {
...
+func FallbackGeneratorNameIfNecessary(
+	generatorName string,
+	resourcesList []*metav1.APIResourceList,
+	cmdErr io.Writer,
+) string {
+	switch generatorName {
+	case DeploymentBasicAppsV1Beta1GeneratorName:
+		if !Contains(resourcesList, appsv1beta1.SchemeGroupVersion.WithResource("deployments")) {
+
+			fmt.Fprintf(cmdErr,
```
Comment (brendandburns) [inline pkg/kubectl/cmd/util/factory_client_access.go]: If you parameterize the API you can turn this into a single utility method for printing the error and use it below. That will clean up this code a bunch.

File: pkg/kubectl/cmd/util/factory_client_access.go [hunk 2]
```diff
@@ -553,6 +556,53 @@ func DefaultGenerators(cmdName string) map[string]kubectl.Generator {
...
+				"WARNING: New deployments generator %q specified, "+
+					"but apps/v1beta1.Deployments are not available. "+
```
Comment (brendandburns) [inline pkg/kubectl/cmd/util/factory_client_access.go]: indent is weird here.

**Comment by brendandburns** [comment]
Generally LGTM, one minor refactoring comment.

Thanks!

**Comment by zjj2wry** [comment]
@brendanburns updated, thank you very much :)

**Comment by zjj2wry** [comment]
@brendanburns ptal

**Comment by xiangpengzhao** [comment]
Please run `./hack/update-bazel.sh`

**Comment by zjj2wry** [comment]
@xiangpengzhao thanks

**Comment by zjj2wry** [comment]
fix issue #51939

**Pull request merged**